include/picongpu/particles/debyeLength/Estimate.kernel

```diff
@@ -0,0 +1,228 @@
+                    }
+                    cupla::__syncthreads(acc);
+                    onlyMaster([&](uint32_t const, uint32_t const) {
+                        auto const mean = supercellSumMomentum / supercellSumWeighting;
+                        // Use a simple variance estimator, the fact it is biased should not matter here
+                        auto const variance = supercellSumMomentumSquared / supercellSumWeighting - mean * mean;
+                        check(acc, electronBox, supercellIdx, supercellSumWeighting, variance, estimateBox);
+                    });
+                }
+
+                /** Check Debye length and write back results
+                 *
+                 * This function makes an estimate for the whole provided electron box,
+                 * and so must be called once per supercell.
+                 *
+                 * @tparam T_Acc alpaka accelerator type
+                 * @tparam T_ElectronBox electron storage type
+                 * @tparam T_ResultBox result storage type, device box for Result
+                 *
+                 * @param acc alpaka accelerator
+                 * @param electronBox electron storage
+                 * @param supercellIdx supercell index, including guards
+                 * @param momentumVariance estimated variance in physical momentums
+                 * @param result device box of type Result with a single element
+                 *        to write kernel result to
+                 */
+                template<typename T_Acc, typename T_ElectronBox, typename T_EstimateBox>
+                DINLINE void check(
+                    T_Acc const& acc,
+                    T_ElectronBox const electronBox,
+                    pmacc::DataSpace<simDim> const supercellIdx,
+                    float_64 const sumWeighting,
+                    float3_64 const momentumVariance,
+                    T_EstimateBox estimateBox) const
+                {
+                    auto const supercellVolume
+                        = CELL_VOLUME * precisionCast<float_64>(SuperCellSize::toRT().productOfComponents());
+                    auto const electronDensity = sumWeighting / supercellVolume;
+
+                    /* For the Maxwell-Boltzmann distribution,
+                     * variance = mass * k * T, both for physical and macroparticle.
+                     * We assume this distribution and estimate k * T out of variance.
+                     * Applying the ideal gas law would result in the same estimate.
+                     */
+                    using Frame = typename T_ElectronBox::FrameType;
+                    auto const mass = static_cast<double>(picongpu::traits::frame::getMass<Frame>());
+                    auto const kT = (momentumVariance[0] + momentumVariance[1] + momentumVariance[2]) / (3.0 * mass);
+                    auto const temperatureKeV = kT * UNIT_ENERGY / UNITCONV_keV_to_Joule;
```

Comment:
Why do you perform the SI conversion on the device?

Reply:
No apparent reason. I think it makes no significant difference in this case, whether to do it inside or aftet the kernel

Reply:
Okay.